### PR TITLE
test: Integration test for db migration

### PR DESCRIPTION
**Pull request by raghuvar-vijay**
Integration test for db migration:
1. The workflow script contains the logic of checking out to the main branch, starting the db scripts with the old schema, and build the AUDITOR cargo package. (Since the main branch does not contain the old schema, for the sake of testing I am checking out to the old commit which still has the old schema. we can change it to main branch after merging this PR)
2. Checkout to the current branch
3. In the test_db_migration.sh script,  we run AUDITOR, insert 2 records and stop.
4. Run sqlx migrate to migrate the db to the new schema.
5. We build the AUDITOR package in the current branch. 
6. Run AUDITOR and insert 2 more records. 
7. Check if 4 records still persists and check the consistency of each record.

### CODE REVIEW

File: .github/workflows/test_db_migration.yml
Comment (dirksammel) [inline .github/workflows/test_db_migration.yml:1]: is there a reason for `runs-on: ubuntu-22.04` instead of `ubuntu-latest`?

Comment (raghuvar-vijay) [reply .github/workflows/test_db_migration.yml:1]: ahh. No reason. Can be changed to latest.

**Review by dirksammel — COMMENTED** [review]
what exactly does the "get current commit" step do in the workflow? I don't think you're using COMMIT_SHA later

**Comment by raghuvar-vijay** [comment]
> what exactly does the "get current commit" step do in the workflow? I don't think you're using COMMIT_SHA later

Yes, it is not used now. I can remove it now. Since I have to anyway refactor after this merge and change the git checkout to main instead of a specific commit.

**Pull request merged**